### PR TITLE
Fix Parquet handling of missing columns

### PR DESCRIPTION
**Pull request by electrum**
The Parquet reader needs to handle the case where columns were added
to the Hive metastore table after the table data was written.

### CODE REVIEW

**Comment by electrum** [comment]
@zhenxiao can you take a look at this?

**Comment by zhenxiao** [comment]
@electrum sure. let me take a look

**Comment by zhenxiao** [comment]
@electrum looks good

File: presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java [hunk 1]
```diff
@@ -415,13 +419,12 @@ public ReadContext init(
...
-                if (!column.isPartitionKey()) {
-                    fields.add(getParquetType(column, messageType));
-                }
-            }
-            MessageType requestedProjection = new MessageType(messageType.getName(), fields.build());
+            List<parquet.schema.Type> fields = columns.stream()
+                    .filter(column -> !column.isPartitionKey())
+                    .map(column -> getParquetType(column, messageType))
+                    .filter(Objects::nonNull)
+                    .collect(toList());
```
Comment (dain) [inline presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java]: use `toImmutableList`

**Comment by dain** [comment]
looks good

**Pull request closed without merging**